lib/pbench/common/__init__.py

```diff
@@ -29,22 +29,32 @@ def wait_for_uri(uri: str, timeout: int):
                   attempts to connect to the URI
 
     Raises:
-        BadConfig : when the URI does not contain either a host or port
+        BadConfig : when the URI is missing the host or when the scheme is
+                    missing or unrecognized
         ConnectionRefusedError : after the timeout period has been exhausted
     """
     url = urlparse(uri)
     if not url.hostname:
         raise BadConfig("URI must contain a host name")
-    if not url.port:
-        raise BadConfig("URI must contain a port number")
+    if url.scheme == "https":
+        port = 443
+    elif url.scheme == "http":
+        port = 80
+    elif not url.scheme:
+        raise BadConfig("URI must include the scheme ('http://' or 'https://')")
+    else:
+        raise BadConfig(f"URI contains an unrecognized scheme, {url.scheme!r}")
+    if url.port:
+        port = url.port
+
     end = time() + timeout
     while True:
         try:
             # The timeout argument to `create_connection()` does not play into
             # the retry logic, see:
             #
             # https://docs.python.org/3.9/library/socket.html#socket.create_connection
-            with socket.create_connection((url.hostname, url.port), timeout=1):
+            with socket.create_connection((url.hostname, port), timeout=1):
```

Comment:
That's too bad: I kinda liked the idea of an `index` `GET` with our expected prefix, since there are possible failure mode not covered by your socket connection. Couldn't get that to work??? 😦

Reply:
It required a lot of unit test changes, and it ended up being _harder_ to diagnose the connection problem, so I reverted to this simple approach and just added the port defaulting.

Reply:
As it turns out, this function also has to support the `postgresql` schema...I don't know whether that supports an HTTP `GET` request...so, it's probably for the best that we stick with the socket connection test.